dali/kernels/test/tensor_test_utils.h

```diff
@@ -0,0 +1,250 @@
+            << "Samples #" << (--max_errors, i)
+            << " have different shapes " << s1[i] << " vs " << s2[i];
+      }
+    }
+  }
+
+  if (errors) {
+    if (errors > max_errors) {
+      max_errors = 0;
+      FAIL() << " (" << (errors - max_errors) << " more)";
+    } else {
+      max_errors -= errors;
+      FAIL();
+    }
+  }
+}
+
+template <int dim1, int dim2>
+void CheckEqual(const TensorListShape<dim1> &s1, const TensorListShape<dim2> &s2) {
+  int max_errors = 100;
+  CheckEqual(s1, s2, max_errors);
+}
+
+
+template <typename T1, typename T2, int dim1, int dim2, typename ElementsOkFunc>
+void Check(
+    const TensorView<StorageCPU, T1, dim1> &tv1,
+    const TensorView<StorageCPU, T2, dim2> &tv2,
+    ElementsOkFunc eq,
+    int &max_errors) {
+  ASSERT_EQ(tv1.shape, tv2.shape)
+      << "Tensors have different shapes "
+      << tv1.shape << " vs " << tv2.shape << "\n";
+
+  ptrdiff_t n = tv1.num_elements();
+  int dim = tv1.dim();
+  TensorShape<-1> pos;
+  pos.resize(dim);
+
+  int errors = 0;
+  for (ptrdiff_t i = 0; i < n; i++) {
+    if (!eq(tv1.data[i], tv2.data[i])) {
+      if (errors++ < max_errors) {
+        EXPECT_TRUE(eq(tv1.data[i], tv2.data[i]))
+          << "Failed at offset " << i << ", pos = " << pos
+          << "tv1[" << i << "] = " << tv1.data[i]
+          << "tv2[" << i << "] = " << tv2.data[i];
+      }
+    }
+
+    for (int j = dim-1; j >= 0; j--) {
+      if (++pos[j] < tv1.shape[j])
+        break;
+      pos[j] = 0;
+    }
+  }
+
+  if (errors) {
+    if (errors > max_errors) {
+      max_errors = 0;
+      FAIL() << " (" << (errors - max_errors) << " more)";
+    } else {
+      max_errors -= errors;
+      FAIL();
+    }
+  }
+}
+
+struct Equal {
+  template <typename T1, typename T2>
+  bool operator()(const T1 &a, const T2 &b) const {
+    return a == b;
+  }
+};
+
+struct EqualEps {
+  EqualEps() = default;
+  explicit EqualEps(double eps) : eps(eps) {}
+
+  template <typename T1, typename T2>
+  bool operator()(const T1 &a, const T2 &b) const {
+    return std::abs(b-a) < eps;
+  }
+  double eps = 1e-6;
+};
+
+template <typename T1, typename T2, int dim1, int dim2, typename ElementsOkFunc = Equal>
+void Check(
+    const TensorView<StorageCPU, T1, dim1> &tv1,
+    const TensorView<StorageCPU, T2, dim2> &tv2,
+    ElementsOkFunc eq = {}) {
+  int max_errors = 100;
+  Check(tv1, tv2, std::move(eq), max_errors);
+}
+
+
+template <typename T1, typename T2, int dim1, int dim2, typename ElementsOkFunc = Equal>
+void Check(
+    const TensorListView<StorageCPU, T1, dim1> &tv1,
+    const TensorListView<StorageCPU, T2, dim2> &tv2,
+    ElementsOkFunc eq = {}) {
+  int max_errors = 100;
+  CheckEqual(tv1.shape, tv2.shape, max_errors);
+  int n = tv1.num_samples();
+  for (int i = 0; i < n; i++) {
+    Check(tv1[i], tv2[i], eq, max_errors);
+  }
+}
+
+
+// FILLING
+
+template <typename Backend, typename T, int ndim>
+struct element_type<TensorView<Backend, T, ndim>> {
```

Comment:
I think this can be added to Tensor[List]View as `using element_type = T`.

Reply:
`element_t<collection>` works for more than just tensors. Adding `element_type` to Tensor[List]View is a good idea, but I didn't add it at the time to avoid merge conflicts with then-pending Kernel API PR. I can add it now.

Reply:
I have no problem with generic `element_t`.